src/libraries/System.Linq/src/System/Linq/Partition.SpeedOpt.cs

```diff
@@ -254,7 +254,7 @@ public TSource[] ToArray()
                 }
 
                 TSource[] array = new TSource[count];
-                for (int i = 0, curIdx = _minIndexInclusive; i != array.Length; ++i, ++curIdx)
+                for (int i = 0, curIdx = _minIndexInclusive; i < array.Length; ++i, ++curIdx)
                 {
                     array[i] = _source[curIdx];
                 }
```

Comment:
```suggestion
                int minIndexInclusive = _minIndexInclusive;
                for (int i = 0; i < array.Length; ++i)
                {
                    array[i] = _source[i + minIndexInclusive];
                }
```
This feels a bit simpler to me.